src/test/java/javaserver/ResponseBuilderTest.java

```diff
@@ -60,30 +51,38 @@ public void testRespondsWith302() {
 
 	@Test
 	public void testFourOhFourResponseCode() throws IOException {
-		ResponseBuilder responder = new ResponseBuilder(unacceptableRequest);
-		Response response = responder.getResponse();
-		assertEquals(response.getResponseCode(), fourOhFour);
+		Response unsupportedRequestResponse = getResponse("GET", "/foo");
+		assertEquals(unsupportedRequestResponse.getResponseCode(), fourOhFour);
 	}
 
 	@Test
 	public void testMethodNotAllowedResponseCode() throws IOException {
-		ResponseBuilder responder = new ResponseBuilder(unallowedRequest);
-		Response response = responder.getResponse();
-		assertEquals(response.getResponseCode(), fourOhFive);
+		Response unallowedRequestResponse = getResponse("POST", "/file1");
+		assertEquals(unallowedRequestResponse.getResponseCode(), fourOhFive);
 	}
 
 	@Test
 	public void testOptionsHeader() throws IOException {
-		ResponseBuilder responder = new ResponseBuilder(requestWithOptions);
-		Response response = responder.getResponse();
-		assertEquals(response.getHeader(), methodOptionsHeader);
+		Response optionsResponse = getResponse("GET", "/method_options");
+		assertEquals(optionsResponse.getHeader(), methodOptionsHeader);
 	}
 
 	@Test
 	public void testRedirectHeader() throws IOException {
-		ResponseBuilder responder = new ResponseBuilder(redirectRequest);
-		Response response = responder.getResponse();
-		assertEquals(response.getHeader(), redirectHeader);
+		Response redirectResponse = getResponse("GET", "/redirect");
+		assertEquals(redirectResponse.getHeader(), redirectHeader);
+	}
+
+	@Test
+	public void testValidCodedParameters200() {
+		Response validParamsResponse = getResponse("GET", codedURI);
+		assertEquals(validParamsResponse.getResponseCode(), twoHundred);
+	}
+
+	@Test
+	public void testInvalidCodedParams404() {
+		Response invalidParamResponse = getResponse("GET", "/foo?");
+		assertEquals(invalidParamResponse.getResponseCode(), fourOhFour);
 	}
```

Comment:
@jayden I've added tests to make sure that the coded parameter string from cob spec receives a 200 and not just any request containing a '?'. My criteria for validity is just to check whether the string "/parameters?" is contained in the GET request, but I'm wondering if that is too specific. Is there a more general rule for what constitutes a valid query with coded parameters?